### PR TITLE
Update getting-started.md

### PR DESCRIPTION
**Pull request by mikemorris**
Documents a few common local dev workflows

### CODE REVIEW

File: dev/docs/getting-started.md [hunk 1]
```diff
@@ -37,3 +37,25 @@ Clean up the gateway you just created:
 ```bash
 kubectl delete -f dev/config/k8s/consul-api-gateway.yaml
 ```
+
+## Deploying a custom Docker image
+
+- Create a Docker image from your local branch with `make docker`
+
+- Upload image to k8s cluster, for a local kind cluster the command is [`kind load docker-image`](https://kind.sigs.k8s.io/docs/user/quick-start/#loading-an-image-into-your-cluster)
```
Comment (andrewstucki) [inline dev/docs/getting-started.md:45]:
```suggestion
- *Optional*: Some k8s setups require loading a custom image into the runtime environment for it to get pulled by worker nodes. For a local kind cluster you can load the image using the subcommand [`kind load docker-image`](https://kind.sigs.k8s.io/docs/user/quick-start/#loading-an-image-into-your-cluster)
```

File: dev/docs/getting-started.md [hunk 2]
```diff
@@ -37,3 +37,25 @@ Clean up the gateway you just created:
 ```bash
 kubectl delete -f dev/config/k8s/consul-api-gateway.yaml
 ```
+
+## Deploying a custom Docker image
```
Comment (andrewstucki) [inline dev/docs/getting-started.md:41]:
```suggestion
## Deploying a custom Docker image on a development cluster
```

**Review by andrewstucki — COMMENTED** [review]
Added some changes since this isn't across the board advice for how to deploy a custom image, just how to do so in a particular development environment.

File: dev/docs/getting-started.md [hunk 3]
```diff
@@ -37,3 +37,25 @@ Clean up the gateway you just created:
...
 kubectl delete -f dev/config/k8s/consul-api-gateway.yaml
 ```
+
+## Deploying a custom Docker image on a development cluster
+
+- Create a Docker image from your local branch with `make docker`
+
+- *Optional*: Some k8s setups require loading a custom image into the runtime environment for it to get pulled by worker nodes. For a local kind cluster you can load the image using the subcommand [`kind load docker-image`](https://kind.sigs.k8s.io/docs/user/quick-start/#loading-an-image-into-your-cluster)
+
+- In `config/deployment/deployment.yaml`, edit `image` to point to the name of the image you uploaded to the cluster.
```
Comment (mikemorris) [inline dev/docs/getting-started.md:47]:
```suggestion
- In [`config/deployment/deployment.yaml`](https://github.com/hashicorp/consul-api-gateway/blob/main/config/deployment/deployment.yaml#L23), edit `image` to point to the name of the image you uploaded to the cluster.
```

**Review by andrewstucki — APPROVED** [review]

**Pull request merged**